WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -170,8 +170,16 @@ def update
     old_confirmation_sent_at = @user.confirmation_sent_at
     if @user.update(user_params)
       if @user.saved_change_to_delegate_status
-        # TODO: See https://github.com/thewca/worldcubeassociation.org/issues/2969.
-        DelegateStatusChangeMailer.notify_board_and_assistants_of_delegate_status_change(@user, current_user).deliver_now
+        if @user.delegate_status
+          if @user.delegate_status == "senior_delegate"
+            @user_senior_delegate = @user
+          else
+            @user_senior_delegate = @user.senior_delegate
+          end
```

Comment:
This might be worth refactoring into a convenience method under `user`:

```ruby
def senior_or_self
  return nil unless self.delegate_status.present?
  
  self.delegate_status == "senior_delegate" ? self : self.senior_delegate
end
```

Reply:
Done.